src/Raven.Server/NotificationCenter/Notifications/Details/RequestLatencyDetail.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.NotificationCenter.Notifications.Details
+{
+    public class RequestLatencyDetail : INotificationDetails
+    {
+        public readonly Dictionary<string, Queue<RequestLatencyInfo>> RequestLatencies = new Dictionary<string, Queue<RequestLatencyInfo>>();
+
+        public void Update(string queryString, long duration, string action)
+        {
+            if (RequestLatencies.TryGetValue(action, out var hintQueue) == false)
+            {
+                var queue = new Queue<RequestLatencyInfo>();
+                queue.Enqueue(new RequestLatencyInfo(queryString, duration, action));
+                RequestLatencies.Add(action, queue);
+            }
+            else
+            {
+                hintQueue.Enqueue(new RequestLatencyInfo(queryString, duration, action));
+            }
+        }
+
+        public DynamicJsonValue ToJson()
+        {
+            var djv = new DynamicJsonValue();
+            foreach (var key in RequestLatencies.Keys)
+            {
+                var queue = RequestLatencies[key];
+                if (queue == null)
+                    continue;
+
+                var list = new DynamicJsonArray();
+                foreach (var details in queue)
+                {
+                    list.Add(details.ToJson());
+                }
+
+                djv[key] = list;
+            }
+
+            return djv;
+        }
+    }
+
+    public struct RequestLatencyInfo : IDynamicJsonValueConvertible
+    {
+        public readonly string QueryString;
```

Comment:
This is no longer a valid way for us to pass queries, we need to have the actual query / params / stuff that can help the user figure this out here.